internal/action/run.go

```diff
@@ -0,0 +1,52 @@
+package action
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/go-tfe"
+	"github.com/sethvargo/go-githubactions"
+)
+
+func Run(inputs Inputs) error {
+	ctx := context.Background()
+
+	client, err := tfe.NewClient(&tfe.Config{
+		Token:   inputs.Token,
+		Address: inputs.Address,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to configure client: %w", err)
+	}
+
+	workspace, err := client.Workspaces.Read(ctx, inputs.Organization, inputs.Workspace)
+	if err != nil {
+		return fmt.Errorf("failed to find workspace %s/%s: %w", inputs.Organization, inputs.Workspace, err)
+	}
+
+	stateVersion, err := client.StateVersions.CurrentWithOptions(ctx, workspace.ID, &tfe.StateVersionCurrentOptions{
+		Include: "outputs",
+	})
+	if err != nil {
+		return fmt.Errorf("failed to fetch state outputs for workspace %s/%s: %w", inputs.Organization, inputs.Workspace, err)
+	}
+
+	for _, o := range stateVersion.Outputs {
+		str := fmt.Sprint(o.Value)
```

Comment:
This seems off, `o.Value` is an` interface{}`, which technically satisfies `Sprint`, but assumes that the concrete value of `Value` is usefully printable. See above, adding more test cases should help.

Reply:
Breaking out an `OutputString(i interface{}) string` func could be helpful. That is the one thing here that would seem to benefit most from some unit testing.